### ISSUE TITLE
Port Recalbox to Odroid c2

### ISSUE THREAD

**Issue by nadenislamarre** [recalbox/recalbox-os#998]
Current status :
- board received, thanks Sylvain ;-)ll
- looks like 64bit is required while only the 64 bits drivers exists (looks like the same for pine64)
- not yet booting

**Comment by JoargTheBard** [comment]
Really interested to see your progress!  I have a c2 as well if you would like help testing

**Comment by Madmatt77** [comment]
Hi !! i have a C2 too and i can help by testing if you need ;) you made a very good job ! i want to put it in my arcade box, I can't wait :D 
Good luck

**Comment by boblablah** [comment]
I realize I am late to the party, but issue #763 was never responded to as far as reasons to do an image for the c1/c1+.  The hardware is greater than the raspi 2 for which the existing recalbox image is maintained.  I am hoping the c1+ becomes a candidate as well for recalbox.

**Comment by nadenislamarre** [comment]
I've finally taked time to see what i can do with this machine i received some monthes ago.
the machine boots the recalbox.
es works.
sound works.
network works.
pads are working.
that's all for the working part.

So, let's see what is not working now :
- ir (infrared) is not working
- the wiimote is not working (patches to be done, like usual)
- kodi doesn't compile
- no emulator compiles

i'm trying to fix it, but what was supposed to be a problem is now a problem : aarch64 architecture is not well supported for the moment.

**Comment by nadenislamarre** [comment]
current status in a 2 minutes video here : http://recalbox.remix.free.fr

**Comment by mfwhite** [comment]
Hello Friend, my name is Michell and I have a Youtube channel called Raspberry Pi BR. In the channel, I analyze systems and boards and I am currently testing the Alpha version of Recalbox to ODROID XU4 and in my opinion is amazing. I wonder if you have future plans for ports to Banana Pi M3 and Orange Pi PC Plus or Plus 2? Cheers from Brazil.

**Comment by nadenislamarre** [comment]
hi,
no ports to Banana Pi M3 and Orange Pi PC Plus or Plus 2.
each port i do add something. i find nothing interesting with these 2 ports. Any arguments to give me ?
Note that it's alpha just because recalbox 4.0 is still alpha. It's working very well.
I'll build a new alpha for x86_64 to let you play the gamecube/wii from the joystick directly.

xu4 port : add the more powerfull board
x86/x86_64: let people to test on their computer without testing
c2 : reduce the price compared to xu4 + may (i hope) let play gamecube while it's arm8 (xu4 is arm7 and dolphin requires x86_64 or arm8 to work)

**Comment by mfwhite** [comment]
Got it, I just thought the Banana Pi M3 could get a Port, being a complete
board (with wifi and Bluetooth), it also has Sata connection and
interesting specifications as eight cores and 2GB of RAM, but unfortunately
not a board that has a lot of support from the developer community. I tried
to test the X86_64 version in my UP Board and although the system was
loaded, the games appeared, but when he opened one game, the screen went
dark and returned to the home screen (any emulator). I appreciate the
attention and answer my questions. Cheers from Brazil.
Em 17/10/2016 06:33, "Nicolas Adenis-Lamarre" <notifications@github.com>
escreveu:

**Comment by nadenislamarre** [comment]
I've release a first alpha for curious testers on http://recalbox.remix.free.fr
It is almost not tested.
However, you can download it while it's still hot.

the delta for things not supposed to work (and so, to not be reported) :
- wiimote classical controller
- MOONLIGHT_EMBEDDED
- QTSIXA_SHANWAN
- PPSSPP (psp)
- MUPEN64PLUS (n64)
- REICAST (dreamcast)
- POCKETSNES (snes9x is however working)
- SCUMMVM
- BEETLE_PCFX
- LUTRO
- UAE (amiga)

**Comment by Madmatt77** [comment]
hi !!
thanks for this release (Oc2) I'll try it tonight  ;)

**Comment by nadenislamarre** [comment]
please note my new comment update :
- hdmi standby after 5 minutes (need a keyboard to get it on again)

it's possibly the same thing that on the xu4 where during 1 second, after minute, the tv becomes black.
However, on the c2, it remains black until you plug a keyboard and press a key.

**Comment by claneys** [comment]
Nice, thanks you. I'll test it as soon as I got the board but it is promising.

Is the standby HDMI can occurs during a movie ? Or it needs completely idle system ?

**Comment by nadenislamarre** [comment]
i don't know. i've not analysed the problem.
It just disable the hdmi output, the system is still responding, but it
seems that somewhere in the c2 code, a keyboard input is not handle like a
joystick input (while on the linux stack, both are events (one with the
flag keyboard to 1)

**Comment by claneys** [comment]
Ok, where c2 code can be found ? (I haven't search for now, it is just to get straight to the goal if you have it near in your memory stack ;) 

I'll try to help if I can since this board is just great ! Hoping to read HEVC video on it and playing game. It will be so perfect.

**Comment by nadenislamarre** [comment]
https://github.com/recalbox/recalbox-buildroot/pull/513
but it should be merged soon.

the main remaining job is to make PPSSPP, MUPEN64PLUS and REICAST working on aarch64.
for PPSSPP, the current version doesn't support aarch64,
but there is currently a pull request that i tested and that allow to compile ppsspp for aarch64. It's not linking, but i'll found why soon. in other word, for ppsspp, it's just a question of time.

for mupen64 and reicast, i'm less optimist for the moment, but i've not search at all.

ir and cec are interesting stuff to do too.

**Comment by nadenislamarre** [comment]
this should help :
http://odroid.com/dokuwiki/doku.php?id=en:c2_turn_off_monitor

**Comment by claneys** [comment]
Ok, so ```setenv "monitor_onoff" "false"``` line into boot.ini should be enough to disabled this feature. Good.

This commit may help RetroPie/RetroPie-Setup#1307

**Comment by Madmatt77** [comment]
Hi !! I downloaded your odroid C2 version of recalbox, put it on SD card (after decompression) but nothing happens when i turn my odroid on
got an idea ?
sorry for my english by the way ... i'm french

**Comment by nadenislamarre** [comment]
you must copy it with dd on linux or win32 image on windows. ill make tuto
for that if not already done by somebody else on the wiki

**Comment by xstrex** [comment]
@nadenislamarre Thanks for all your hard work on getting Recalbox working on the Odroid C2. I've been trying to test it out on my c2, but I too am running into some problems booting the image. I've downloaded, uncompressed and wrote the image via dd, to the emmc, attached it to the c2, and powered it up. I get a red power light, but no blue light (which would normally be solid as it loads the kernel). The display connected to the HDMI never turns on. I've also connected the c2 to a serial cable to monitor the boot process, and here's what I get: 

`GXBB:BL1:08dafd:0a8993;FEAT:EDFC318C;POC:3;RCY:0;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:1;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:2;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:3;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:4;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:5;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:6;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:7;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:8;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:9;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:A;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:B;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:C;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:D;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:E;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:F;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;LOOP:10;EMMC:0;READ:0;CHK:F3;SD:800;USB:8;`

From what I can tell, the kernel never loads. 

Do you have any advice on how to get this image to boot properly? 

I don't want to jump to the conclusion that the image is corrupt or anything, but I've written thousands of images to media via dd over the years, and only seen this problem when the image is corrupt, or the partition table is incomplete, or misconfigured in some way. 

Any advice you can give would be greatly appreciated. Thanks again for doing this.

**Comment by xstrex** [comment]
So I've managed to write your image to an sd card and that booted fine, but I've run into another issue. I can't seem to get a ps3 controller to pair or even connect. It almost looks like ps3 support is missing entirely, or perhaps there's some udev rules missing? Here's the output of /var/log/messages when the controller is plugged in to the c2: 

> Oct 26 05:43:25 RECALBOX user.err kernel: [  216.170740] usb 1-1.3: device descriptor read/8, error -71
> Oct 26 05:43:25 RECALBOX user.warn kernel: [  216.303595] ERROR::handle_hc_nak_intr:1307: Can not read device info from hub.We take it error
> Oct 26 05:43:25 RECALBOX user.warn kernel: [  216.303595]
> Oct 26 05:43:25 RECALBOX user.warn kernel: [  216.306987] ERROR::handle_hc_nak_intr:1307: Can not read device info from hub.We take it error
> Oct 26 05:43:25 RECALBOX user.warn kernel: [  216.306987]
> Oct 26 05:43:25 RECALBOX user.warn kernel: [  216.310439] ERROR::handle_hc_nak_intr:1307: Can not read device info from hub.We take it error
> Oct 26 05:43:25 RECALBOX user.warn kernel: [  216.310439]

Suggestions? Thanks.

**Comment by nadenislamarre** [comment]
i'll do some checks with my ps3 controllers.

Good news, i've found how to remove the black screen that occurs exactly after 10 minutes : add consoleblank=0 into the boot.ini file on kernel parameters.

double good news : this looks like the same thing on the odroid xu4 (except that the black screen happen only 1 sec) while it occurs exactly at 10 minutes.

This is the tty "screensaver" that occurs by default at 10 minutes ;-)

to test :
(cat /sys/module/kernel/parameters/consoleblank)

**Comment by xstrex** [comment]
@nadenislamarre I can confirm that adding consoleblank=0 to the end of the kernel parameters works! Tested on Odroid c2.

**Comment by xstrex** [comment]
For anyone wondering what this looks like, here's my kernel line from /boot/boot.ini:
`setenv bootargs "console=tty2 root=/dev/mmcblk0p2 rootwait ro ${cec} quiet loglevel=0 consoleblank=0"`

**Comment by Madmatt77** [comment]
Thanks it works !! but i can't connect a ps3 controller right now. is there something to do ?

**Comment by nadenislamarre** [comment]
i confirm the issue with the ps3 pad :
[  165.813974] input: Sony PLAYSTATION(R)3 Controller as /devices/platform/dwc2_b/usb1/1-1/1-1.2/1-1.2:1.0/0003:054C:0268.0002/input/input3
[  165.814450] sony 0003:054C:0268.0002: input,hiddev0: USB HID v1.11 Joystick [Sony PLAYSTATION(R)3 Controller] on usb-dwc2_b-1.2/input0
[  165.818375] ERROR::handle_hc_nak_intr:1307: Can not read device info from hub.We take it error

**Comment by nadenislamarre** [comment]
cec is implemented as a keyboard on the odroid c2.
unfortunally, most of my remotes key doesn't work ;-(

can somebody do :
- run "evtest" from ssh, choose the cec_input entry, and check that cec for him too. (and confirm that up/down/left/right keys work)
- confirm that the tv remote control allows to control emulationstation

**Comment by nadenislamarre** [comment]
embeded infrared device is now working for the next build,
you can follow https://github.com/recalbox/recalbox-os/wiki/Infra-red-remote-control-on-the-Recalbox-%28EN%29 and take any remote control to use with kodi.

**Comment by nadenislamarre** [comment]
for the next release, i've added in the boot.ini some parameters to control the cpu power.
if you've performances troubles with an emulator, don't hesitate to lower the number of cpu and increase the power of the remaining cpu and make me a feedback.

from the official odroid c2 wiki :

# 1.680 GHz 	"1680" 	booting OK with 4 cores / recommend 3 cores for heavy load test
# 1.752 GHz 	"1752" 	booting OK with 4 cores / recommend 2 cores for heavy load test
# 1.896 GHz 	"1896" 	booting OK with 1~3 cores / recommend running with a single core
# 1.920 GHz 	"1920" 	booting OK with 1 or 2 cores / recommend running with a single core
# 1.944 GHz 	"1944" 	booting OK with a single core but unstable
# 2.016 GHz 	"2016" 	extremely unstable

# max cpu frequency in dvfs table / in MHz unit
# setenv max_freq "2016"  # 2.016GHz
# setenv max_freq "1944"  # 1.944GHz
# setenv max_freq "1944"  # 1.944GHz
# setenv max_freq "1920"  # 1.920GHz
# setenv max_freq "1896"  # 1.896GHz
# setenv max_freq "1752"  # 1.752GHz
# setenv max_freq "1680"  # 1.680GHz
# setenv max_freq "1656"  # 1.656GHz
# setenv max_freq "1536"  # 1.536GHz
setenv max_freq "1536"

# max cpu-cores
# setenv maxcpus "1"
# setenv maxcpus "2"
# setenv maxcpus "3"
# setenv maxcpus "4"
setenv maxcpus "4"

**Comment by xstrex** [comment]
Thanks @nadenislamarre Do you still need someone to test the cec with evtest?

**Comment by nadenislamarre** [comment]
@xstrex yes please.

**Comment by xstrex** [comment]
@nadenislamarre testing now.

**Comment by nadenislamarre** [comment]
come on https://kiwiirc.com/client/irc.freenode.net/#recalbox please

**Comment by xstrex** [comment]
@nadenislamarre evtest output: 
# evtest /dev/input/event0
Input driver version is 1.0.1
Input device ID: bus 0x10 vendor 0x1b8e product 0xcec version 0x1
Input device name: "cec_input"
Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 2 (KEY_1)
    Event code 3 (KEY_2)
    Event code 4 (KEY_3)
    Event code 5 (KEY_4)
    Event code 6 (KEY_5)
    Event code 7 (KEY_6)
    Event code 8 (KEY_7)
    Event code 9 (KEY_8)
    Event code 10 (KEY_9)
    Event code 11 (KEY_0)
    Event code 28 (KEY_ENTER)
    Event code 52 (KEY_DOT)
    Event code 103 (KEY_UP)
    Event code 105 (KEY_LEFT)
    Event code 106 (KEY_RIGHT)
    Event code 108 (KEY_DOWN)
    Event code 113 (KEY_MUTE)
    Event code 114 (KEY_VOLUMEDOWN)
    Event code 115 (KEY_VOLUMEUP)
    Event code 116 (KEY_POWER)
    Event code 128 (KEY_STOP)
    Event code 139 (KEY_MENU)
    Event code 158 (KEY_BACK)
    Event code 161 (KEY_EJECTCD)
    Event code 163 (KEY_NEXTSONG)
    Event code 164 (KEY_PLAYPAUSE)
    Event code 165 (KEY_PREVIOUSSONG)
    Event code 166 (KEY_STOPCD)
    Event code 167 (KEY_RECORD)
    Event code 168 (KEY_REWIND)
    Event code 172 (KEY_HOMEPAGE)
    Event code 174 (KEY_EXIT)
    Event code 200 (KEY_PLAYCD)
    Event code 201 (KEY_PAUSECD)
    Event code 208 (KEY_FASTFORWARD)
    Event code 226 (KEY_MEDIA)
    Event code 256 (BTN_0)
    Event code 362 (KEY_PROGRAM)
    Event code 363 (KEY_CHANNEL)
    Event code 366 (KEY_PVR)
    Event code 386 (KEY_TUNER)
    Event code 398 (KEY_RED)
    Event code 399 (KEY_GREEN)
    Event code 400 (KEY_YELLOW)
    Event code 401 (KEY_BLUE)
    Event code 402 (KEY_CHANNELUP)
    Event code 403 (KEY_CHANNELDOWN)
    Event code 765 (?)
Properties:
Testing ... (interrupt to exit)

Unfortunatly my remote does not trigger any events, i've verified that cec passthrough is enabled on my receiver. and tried both the receiver remote, and my universal remote (which both use IR), but they are unable to control emulationstation. Testing Kodi now.

**Comment by xstrex** [comment]
@nadenislamarre I'll try the emmc tests you mentioned in the next few days. Thank you.

**Comment by xstrex** [comment]
@nadenislamarre It would be great if we could get ps3 controllers working, I know a few of us (myself included) use that as the sole controller for the system. Without it, we're left with just a keyboard. Thanks.

**Comment by xstrex** [comment]
@nadenislamarre I've done some tests with the emmc.. 
1. burned ubuntu64 16.04 from: http://odroid.in/ubuntu_16.04lts/ to emmc. 
2. installed in c2, booted normally. 
3. moved /boot contents to computer
4. moved recalbox /boot contents to /boot on emmc
5. booted c2 with emmc, booted up fine. 
6. logged into c2 via ssh. c2 running ubuntu64 (because only the /boot contents were changed). 

On my system if I mount the recalbox.img, I get 3 partitions, disk2s0, disk2s1 & disk2s2. Somehow I’ve got to get the contents of disk2s2 written to the disk2s2 on the emmc. 

Here’s the output from diskutil on a mac showing the partitions of the recalbox.img:
```
/dev/disk2 (disk image):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:     FDisk_partition_scheme                        +1.2 GB     disk2
   1:             Windows_FAT_32 BOOT                    67.1 MB    disk2s1
   2:                      Linux                         1.1 GB     disk2s2
```

Is it possible to take the ubuntu64 image from above, replace the /boot contents with that from recalbox/boot, and the 2’nd partition with the contents from recalbox? But leave everything else intact? I’m a bit at a loss as to what to do next. Any advice?

**Comment by nadenislamarre** [comment]
See :
https://github.com/recalbox/recalbox-buildroot/blob/rb-4.1.X/board/recalbox/copy-recalbox-archives.sh#L48

the important is to not break the beginning of the card.
MBR is 512 bytes basically for the partition table.
bl1 is a binary specific to the c2 done by hardkernel to boot the c2 to be able to boot on uboot
uboot is able to load the kernel on the BOOT partition
BOOT is able to mount the recalbox partition and load the recalbox system.

my guess is that my bl1 doesn't support emmc. Perhaps i've not the latest one.
What you can do :

1) mount the Linux partition, erase all the files, and copy the one from the recalbox sdcard.
check that recalbox starts on the emmc.

2) dd your emmc card to me (only the first 2 mb)
something like dd if=/dev/mmblk0 of=/tmp/emmc count=1300
and send me the emmc file.

3) if you're motivated enough :
here you can find what i do with the bl1 and uboot : https://github.com/recalbox/recalbox-buildroot/blob/rb-4.1.X/board/recalbox/copy-recalbox-archives.sh#L68
i take the 2 files from here :
https://github.com/recalbox/recalbox-buildroot/blob/rb-4.1.X/configs/recalbox-odroidc2_defconfig#L249
so, in other words, from here : https://github.com/hardkernel/u-boot/tree/odroidc2-v2015.01/sd_fuse

the question is :
does ubuntu uses the same bl1 ? (can be checked with dd the 2 parts and diff them).
If no, where does ubuntu take the bl1 file ?
If yes, is there a uboot configuration specific to the emmc ?

**Comment by nadenislamarre** [comment]
psp is now working (but the controls are broken ; to be fixed, but it' sjust a question of time)
n64 is now working.

**Comment by xstrex** [comment]
@nadenislamarre As per your guidance I’ve removed the emmc & sd cards, mounted both of them on a separate linux box and wiped out the contents of the third partition (sdb2) with rm -rf *, and copied the contents from the second partition on the SD to the partition on the emmc with rsync. Effectively cloning the system partition on to the system partition of the emmc. 

Unmounted the emmc, attached it to the c2, and booted it up (without the sd attached). 
This time the c2 booted right up, loaded the kernel, mounted root fs, started user land services etc. After a short while it was ping-able, and I could get to the rommanager page via http! so it worked.. 
But I could Not login via ssh.. and samba access is down. The reason is, Ubuntu only has 2 partitions, /boot and /, Where as recalbox has 3, /boot, / & user data partition (listed here: https://github.com/recalbox/recalbox-os/wiki/Recalbox-development---basics-%28EN%29), it's my understanding that this last one should be created automatically on first boot, but perhaps it isn't since this isn't the first boot, and more of a franken-emmc install. 

Looks like progress to me! 

I'll get a dd of the first part of the emmc to you now. How, or where would you like it?

**Comment by nadenislamarre** [comment]
the 3rd partition on recalbox is not mandatory. If recalbox cannot create it, it create it in memory.
i guess you did not copy correctly (because i didn't tell you exactly how).
Anyway, just send me your 2mB boot sector via email for example (nicolas dot adenis dot lamarre at gmail dot com).
After that, i'll tell you how to convert the sdcard.img to an emmc.img (basically, by doing dd if=/path/to/your/emmc.boot.you.will.send.me of=sdcard.img count=X skip=Y seek=Y bs=4M). X and Y have to be determined.

**Comment by xstrex** [comment]
@nadenislamarre I'm positive that I copied it correctly, there must be something else going on, (I used rsync -aHAXv /media/recal/ /media/ubuntu/) this isn't my first rodeo, hehe. 
Anyways, I'll send you the boot img I created.

**Comment by xstrex** [comment]
@nadenislamarre Found one problem, fixing it now to see if it fixes the issue. 
```
/dev/sda1        63M   14M   50M  21% /media/ubuntu/1
/dev/sda2       1.1G  589M  384M  61% /media/ubuntu/2
/dev/sda3        28G  995M   26G   4% /media/ubuntu/3
/dev/sdb1       128M   29M  100M  23% /media/recal/1
/dev/sdb2        15G  671M   13G   5% /media/recal/2
/dev/sdb3       495K   10K  450K   3% /media/recal/3
```
SD card with recalbox is sdb (partitions 1, 2 & 3)
Emmc with ubuntu formally is sda (partitions 1, 2 & 3)
1 = boot, 2 = root, 3 = userdata
Partition sizes are wrong, I'm going to resize them and re-rsync the data from the SD to the emmc.

**Comment by xstrex** [comment]
@nadenislamarre ...and fixed. I've got your custom recalbox image booting off the emmc!
Here's some details:
```
# uname -a
Linux RECALBOX 3.14.79 #1 SMP PREEMPT Mon Oct 17 08:03:39 CEST 2016 aarch64 GNU/Linux
```
```
# cat /etc/os-release
NAME=Buildroot
VERSION=2016.05-git-gfe2dd00-dirty
ID=buildroot
VERSION_ID=2016.05-git
PRETTY_NAME="Buildroot 2016.05-git"
```
```
# df -h
Filesystem                Size      Used Available Use% Mounted on
/dev/root               984.9M    665.3M    262.7M  72% /
devtmpfs                736.3M         0    736.3M   0% /dev
tmpfs                   858.7M         0    858.7M   0% /dev/shm
tmpfs                   858.7M      4.0K    858.7M   0% /tmp
tmpfs                   858.7M      2.8M    855.9M   0% /var
tmpfs                   858.7M    108.0K    858.6M   0% /run
/dev/mmcblk0p1          127.7M     28.2M     99.5M  22% /boot
/dev/mmcblk0p3           13.4G      2.3M     12.8G   0% /recalbox/share
```

**Comment by nadenislamarre** [comment]
dd if=output/images/bl1.bin.hardkernel of=mine.bl1.96 skip=1
dd if=emmc-testing.img of=ubuntu.bl1.bin.hardkernel_96 skip=1 count=96
md5sum mine.bl1.96 ubuntu.bl1.bin.hardkernel_96
73d667c9b99c3c1dac22992b5a8ff793  mine.bl1.96
73d667c9b99c3c1dac22992b5a8ff793  ubuntu.bl1.bin.hardkernel_96

it looks like that the bl1 are the same.

so my guess is that the uboot options are different.
can you do the following :

1) burn my alpha1 c2 onto your emmc card
2) verify it doesn't boot
3) get the uboot from ubuntu : dd if=emmc-testing.img of=ubuntu.uboot skip=97 count=1184
4) replace my uboot by the ubuntu one : dd if=ubuntu.uboot of=/dev/rdiskthegoodone seek=97
5) and try to boot to confirm that the problem is ONLY in uboot.

**Comment by nadenislamarre** [comment]
>>> remaining job :
    overclock via the interface
    not working bluetooth : ps3 / wiimote classical controller
    emmc card
    MOONLIGHT_EMBEDDED
    QTSIXA_SHANWAN
    PPSSPP (psp) (controls not working)
    REICAST (dreamcast)
    POCKETSNES (snes9x is however working)
    SCUMMVM
    BEETLE_PCFX
    LUTRO

>>> working for the alpha2 :
- n64, psp (keyboard controls), lirc
- blackscreen removed
- overclock options in the .ini
- virtual keyboard in es for the wifi for example

**Comment by nadenislamarre** [comment]
alpha2 released.

**Comment by nadenislamarre** [comment]
@xstrex i've found an emmc option for uboot which could be the good one. Can you try :

1) download my c2 alpha2
2) wget http://recalbox.remix.free.fr/special/u-boot.bin.gz
3) gunzip u-boot.bin.gz
4) dd if=u-boot.bin of=sdcardc2alpha2.img seek=97 conv=notrunc
5) try to boot

**Comment by xstrex** [comment]
@nadenislamarre sounds good, I'll give it a shot tomorrow. 

Also I recently got a XU4. DD'd Ubuntu on the emmc, then used gparted to resize the partitions, mounted them and removed their contents. Then extracted your latest XU4 boot & rootfs to them. Plugged it in the XU4, and it booted up first try! It's been working great! Everything works so far. Even with a PS3 controller (though the lights on the controller never stop flashing, but it works just fine)!

**Comment by smeierger** [comment]
Thank you for your work and time!! 
N64 is working! Great Job.
But the Sega32x doesnt work. It is incredible slow and the Xbox360 Controllers dont work in this Emu

**Comment by nadenislamarre** [comment]
BEETLE_PCFX => not supported
LUTRO => not supported
POCKETSNES (snes9x is however working) => not supported

- MOONLIGHT_EMBEDDED => done for alpha3
- QTSIXA_SHANWAN => done for alpha3
- SCUMMVM => done for alpha3

remaining job :
- overclock via the interface
-not working bluetooth : ps3 / wiimote classical controller
- emmc card
- REICAST (dreamcast) => probably not supported
-PPSSPP (psp) (controls not working)

**Comment by xstrex** [comment]
@nadenislamarre 
burned alpha2 to emmc with dd.. System will not boot. 
burned uncompressed new u-boot.bin to emmc… System will not boot.

**Comment by nadenislamarre** [comment]
and burn just the extracted ubuntu u-boot.bin ?

**Comment by xstrex** [comment]
What do you mean, just the extracted u-boot.bin? I've got a few, can you meet me here: https://kiwiirc.com/client/irc.freenode.net/#recalbox

**Comment by Madmatt77** [comment]
Hi !! thanks for the alpha2 !!
before buying a x360 controller, does it work with ?
same question with a ps3 controller connected by usb ?
any chances reicast works correctly one day ?
thanks ;)

**Comment by smeierger** [comment]
The Xbox 360 Controller work, but not in all emulators. I think it is a configuration problem, 
but more tests need. :-)

Am 02.11.2016 um 17:35 schrieb Madmatt77:

**Comment by nadenislamarre** [comment]
@xstrex 
i guess that uboot is not correct on my c2 build.
i want to confirm that point by replacing the uboot ubuntu binary on the recalbox.img

1) download the c2 alpha2 image : recalbox.img
2) extract the ubuntu uboot binary : dd if=emmc-testing.img of=ubuntu.uboot skip=97 count=1184
3) update my c2 alpha2 image with the ubuntu uboot binary : dd if=ubuntu.uboot of=recalbox.img seek=97 conv=notrunc
4) burn this new image on the emmc and try to boot.

**Comment by nadenislamarre** [comment]
about the ps3 controller :
odroid c2 kernel is version 3.14.
upgrading the driver to version 3.15 is not possible because a lot of work was done on hid (human interface device drivers) ; it doesn't compile and complains about inexisting new hid functions.
i've downgrade to version 3.13. It compiles, but i get the same issue.
Googling this error returns only some links, and all related to the odroid. My guess for the moment is that an odroid patch breaks something related to hid ;-(

**Comment by nadenislamarre** [comment]
https://www.google.fr/?gws_rd=ssl#q=+%22We+take+it+error%22

**Comment by nadenislamarre** [comment]
something funny.
i spoke to a collegue about odroids some weeks ago (he has one).
Today he wrote me about a new linux distribution called "recalbox" that is now supported on odroids to ask me if i had ever tested it.

**Comment by nadenislamarre** [comment]
@xstrex do you have a powered usb hub ? a #odroid irc people tells it happends when the usb device eats too voltage. Strange however while i've just that plugged. I've however no way to test. Can you test this point ?

**Comment by nadenislamarre** [comment]
remaining job :
- overclock via the interface

suspended:
- emmc card => required help from xstrex
- ppsspp => controls are beeing fixed by subs
-not working bluetooth : ps3 / wiimote classical controller => to be tested with a powered usb (source = #odroid)

**Comment by xstrex** [comment]
@nadenislamarre I'll work on this more today, I've been pretty busy with work lately. Will you be around?

**Comment by xstrex** [comment]
@nadenislamarre Tried these steps, and it did not boot:
1) download the c2 alpha2 image : recalbox.img
2) extract the ubuntu uboot binary : dd if=emmc-testing.img of=ubuntu.uboot skip=97 count=1184
3) update my c2 alpha2 image with the ubuntu uboot binary : dd if=ubuntu.uboot of=recalbox.img seek=97 conv=notrunc
4) burn this new image on the emmc and try to boot.

I'm not sure I understand your question about the usb hub. I do use a usb hub, but i've written hundreds of images using it and never had an issue related to power. Besides if it was an issue with the hub, then why would burning hardkernel's ubuntu image to the emmc work just fine?

**Comment by claneys** [comment]
@nadenislamarre : I tested too without success with ubuntu uboot and your recalbox uboot.

**Comment by hgomezsa** [comment]
Frontend works but I can confirm ps3 dualshock doesnt work. If it runs on top of linux, I was able to compile a dualshock driver and pair the gamepad with a usb bt dongle on ubuntu. Don't know if I still keep the binaries but.

BTW can you execute any game on mame/fba? I wasn't able to do it with retroarch mame/fba cores on android emulator...

Hope you doesn't abandon this project. Interface is really fast on C2 (I have comared it to stable 4.0 on an extreme oced raspberry pi 1)

**Comment by substring** [comment]
@hgomezsa just tried a shanwan clone. When i plug it on USB, i get constant errors :
```
[  300.001873] input: ShanWan PS(R) Ga`epad as /devices/platform/dwc2_b/usb1/1-1/1-1.3/1-1.3:1.0/0003:054C:0268.0013/input/input22
[  300.002168] sony 0003:054C:0268.0013: input,hiddev0: USB HID v1.10 Joystick [ShanWan PS(R) Ga`epad] on usb-dwc2_b-1.3/input0
[  300.009352] usb 1-1.3: input irq status -5 received
[  301.214175] usb 1-1.3: USB disconnect, device number 71
[  301.530837] usb 1-1.3: new full-speed USB device number 72 using dwc_otg
[  301.631557] usb 1-1.3: unable to read config index 0 descriptor/start: -32
[  301.631564] usb 1-1.3: chopping to 0 config(s)
[  301.632301] usb 1-1.3: string descriptor 0 read error: -32
[  301.632544] usb 1-1.3: no configuration chosen from 0 choices
```
My xarcade doesn't work either. I believe it's a major kernel problem here (kernel 3.14 is pretty outdated, but hardkernel is not doing anything to improve that)

**Comment by Madmatt77** [comment]
Hi !!
i have a question about your batocera project for odroid c2 : 
i want to conect a panel with gpio to my odroid c2, how can io do that ? i'm very newbee ...
do i need to configure somthing or it's only connection ?
thanks and sorry for my bad english ..

**Comment by substring** [comment]
solved with https://github.com/recalbox/recalbox-os/releases/tag/4.1.0
Images available at https://archive.recalbox.com
